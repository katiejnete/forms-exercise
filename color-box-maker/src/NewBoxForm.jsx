import React, {useState} from "react";
import {Formik, Form, Field} from 'formik';

const NewBoxForm = ({addBox}) => {
const INITIAL_STATE = {
    backgroundColor: "",
    width: "",
    height: ""    
};
const [formData, setFormData] = useState(INITIAL_STATE);
const handleChange = (evt) => {
    const {name, value} = evt.target;
    setFormData(fData => ({
        ...fData,
        [name]: value
    }));
}
const handleSubmit = (evt) => {
    evt.preventDefault();
    addBox(formData);
    setFormData(INITIAL_STATE);
};

return (
    <form onSubmit={handleSubmit}>
        <label htmlFor="backgroundColor">Background color:</label>
        <input 
        id="backgroundColor"
        type="text"
        name="backgroundColor"
        value={formData.backgroundColor}
        onChange={handleChange}
         />
        <label htmlFor="width">Width:</label>         
        <input 
        id="width"
        type="text"
        name="width"
        value={formData.width}
        onChange={handleChange}
         />
        <label htmlFor="height">Height:</label>         
        <input 
        id="height"
        type="text"
        name="height"
        value={formData.height}
        onChange={handleChange}
         />
        <button>Add box</button>
    </form>
)
};

export default NewBoxForm;