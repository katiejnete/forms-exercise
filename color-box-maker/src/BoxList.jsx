import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import { v4 as uuid } from "uuid";

const BoxList = () => {
  const INITIAL_STATE = [];
  const [boxes, setBoxes] = useState(INITIAL_STATE);
  const addBox = (box) => {
    const newBox = { ...box, id: uuid() };
    setBoxes((boxes) => [...boxes, newBox]);
  };
  function deleteBox(id) {
    setBoxes(boxes.filter((box) => box.id !== id));
  }

  return (
    <>
      <NewBoxForm addBox={addBox} />
      {boxes.map(({ id, backgroundColor, width, height }) => (
          <Box
            key={id}
            id={id}
            backgroundColor={backgroundColor}
            width={width}
            height={height}
            deleteBox={deleteBox}
          />
        ))}
    </>
  );
};

export default BoxList;
