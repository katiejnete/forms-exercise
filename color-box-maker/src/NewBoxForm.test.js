import React from "react";
import {render} from "@testing-library/react";
import NewBoxForm from "./NewBoxForm.jsx";
import {addBox} from "./BoxList.jsx"

// smoke test
it("renders without crashing", () => {
    render(<NewBoxForm addBox={addBox} />);
})

// snapshot test
it("matches snapshot", () => {
    const {asFragment} = render(<NewBoxForm addBox={addBox} />);
    expect(asFragment()).toMatchSnapshot();
})