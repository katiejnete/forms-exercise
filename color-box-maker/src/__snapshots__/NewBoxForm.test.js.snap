// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="backgroundColor"
    >
      Background color:
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      type="number"
      value=""
    />
    <button
      type="submit"
    >
      Add box
    </button>
  </form>
</DocumentFragment>
`;
